File: problem/Concatenation_Consecutive/concatenatedBinary.c
int concatenatedBinary(int n)
{
    int MOD = 1000000007;
    long long res = 0;
    int bits = 0;

    for (int i = 1; i <= n; i++)
    {
        if ((i & (i - 1)) == 0)
            bits++;
        res = ((res << bits) | i) % MOD;
    }

    return res;
}

File: problem/Concatenation_Consecutive/concatenatedBinary.js
/**
 * @param {number} n
 * @return {number}
 */
var concatenatedBinary = function(n) {
    const MOD = 1000000007n;
    let res = 0n;
    let bits = 0n;

    for(let i=1n; i<=BigInt(n); i++){
        if((i & (i-1n)) === 0n) bits++;
        res = ((res<<bits) | i) % MOD;
    }

    return Number(res);
};